notebooks/stac-load-S2-deafrica.py

```diff
@@ -0,0 +1,154 @@
+# ---
+# jupyter:
+#   jupytext:
+#     formats: ipynb,py:percent
+#     text_representation:
+#       extension: .py
+#       format_name: percent
+#       format_version: '1.3'
+#       jupytext_version: 1.13.6
+#   kernelspec:
+#     display_name: 'Python 3.8.12 64-bit (''stac'': conda)'
+#     language: python
+#     name: python3
+# ---
+
+# %% [markdown]
+# # Access Sentinel 2 Analysis Ready Data from Digital Earth Africa
+#
+# [![Binder](https://mybinder.org/badge_logo.svg)](https://mybinder.org/v2/gh/opendatacube/odc-stac/develop?labpath=notebooks%2Fstac-load-S2-deafrica.ipynb)
+#
+# https://explorer.digitalearth.africa/products/s2_l2a
+
+# %% [markdown]
+# ## Import required packages
+
+# %%
+import rasterio
+from pystac_client import Client
+
+from odc.stac import stac_load
+
+# %% [markdown]
+# ## Set configuration
+#
+# The configuration dictionary is determined from the product's definition, availble at https://explorer.digitalearth.africa/products/s2_l2a#definition-doc
+#
+# All assets except SLC have the same configuration. SLC uses `uint8` rather than `uint16`.
+#
+# In the configuration, we also supply the aliases for each band. This means we can load data by band name rather than band number.
+
+# %%
+config = {
+    "s2_l2a": {
+        "assets": {
+            "*": {
+                "data_type": "uint16",
+                "nodata": 0, 
+                "unit": "1",
+            },
+            "SLC": {
+                "data_type": "uint8",
+                "nodata": 0, 
+                "unit": "1",
+            },
+        },
+        "aliases": {
+            "costal_aerosol": "B01",
+            "blue": "B02",
+            "green": "B03",
+            "red": "B04",
+            "red_edge_1": "B05",
+            "red_edge_2": "B06",
+            "red_edge_3": "B07",
+            "nir": "B08",
+            "nir_narrow": "B08A",
+            "water_vapour": "B09",
+            "swir_1": "B11",
+            "swir_2": "B12",
+            "mask": "SLC",
+            "aerosol_optical_thickness": "AOT",
+            "scene_average_water_vapour": "WVP",
+        }
+    }
+}
```

Comment:
Not really a request to change but an "explainer". Ideally this information would come directly from STAC.
Per band `data-type`/`nodata`/`units` is covered by "Raster Extension", and "aliases" are covered by "EO" extension (`common_name`) is how they call it. Would be nice is if explorer included that information, at least the Raster part (common name is a bit harder as datacube allows several aliases). This information is already present in the datacube, and should not be too hard to expose. Not sure if this is in the roadmap @alexgleith?

more info here:
https://odc-stac.readthedocs.io/en/latest/stac-best-practice.html

Reply:
Agreed, would be great if the user didn't have to specify this.